File: EnforcerServices/src/main/java/com/webforged/enforcer/management/data/AllowedArtifact.java
package com.webforged.enforcer.management.data;

import java.time.Instant;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Table;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
@Table("allowedartifacts")
public class AllowedArtifact {
	
	@Id
	public Long allowed_artifact_id ;
	private Long artifact_id;
	private Long project_id;
	private String approval_architect;
	private Instant approval_ts;
	
	public AllowedArtifact() {
	}
}

File: EnforcerWeb/src/app/model/AllowedArtifact.ts
export class AllowedArtifact {
	public allowed_artifact_id: number;
	public project_id: number;
	public artifact_id: number;
	public approval_architect?: string;
	public approval_ts?: Date;

	constructor( obj: Object = {}) {
		Object.assign(this, obj) ;
	}
 
	public toJSON() {
		return {
			allowed_artifact_id: this.allowed_artifact_id,
			project_id: this.project_id,
			artifact_id: this.artifact_id,
			approval_architect: this.approval_architect,
			approval_ts: this.approval_ts,
		}
	}
}